charts/spire/charts/spire-agent/templates/configmap.yaml

```diff
@@ -1,3 +1,5 @@
+{{- include "spire-lib.strict-check" (list . "clusterName must be set" (eq (include "spire-lib.cluster-name" .) "example-cluster"))}}
```

Comment:
We should not provide a default cluster name or trust domain.

Out of all of the settings, these should be specified by the user 100% of the time.

Reply:
What about macro's suggestion of still keeping the ability to easily kick the tires, but turn it off by default?

Reply:
I'm willing to be flexible, but I really don't like the precedent that we either have already set, or are committing to setting.

If a installation using our Helm charts isn't production quality, then we should fix it to be production quality, adjust what we decide is production quality, or adjust our documentation to indicate that initially the installation won't be production quality, but after you set a specific property, it will be?

Personally I don't care which one is the path forward; and in the interest of getting some movement, and @marcofranssen call that we easily "kick the tires, but turn (I'm not sure if it is "production quality" or "easy non-production prototype installations", I'm going to withdraw my objection and clear this for the other maintainers to move in whatever direction they deem fit.